synapse/handlers/presence.py

```diff
@@ -724,12 +739,10 @@ async def _update_states(self, new_states: Iterable[UserPresenceState]) -> None:
                     self.state,
                 )
 
-                # Since this is master we know that we have a federation sender or
-                # queue, and so this will be defined.
-                assert self._federation
-
                 for destinations, states in hosts_and_states:
-                    self._federation.send_presence_to_destinations(states, destinations)
+                    self._federation_queue.send_presence_to_destinations(
+                        states, destinations
+                    )
```

Comment:
@richvdh To continue the conversation from https://github.com/matrix-org/synapse/pull/9828#discussion_r614987677, this is a similar shape as in `maybe_send_presence_to_interested_destinations`, except it uses the federation queue instead of the federation sender. I don't know whether it is worth trying to merge them, but I think that gets a bit convoluted?